rust/src/detect/iprep.rs

```diff
@@ -112,6 +112,28 @@ pub fn detect_parse_iprep(i: &str) -> IResult<&str, DetectIPRepData, RuleParseEr
             mode,
         };
         return Ok((i, DetectIPRepData { du8, cat, cmd }));
+    } else if values.len() == 3 {
+        let cmd = match DetectIPRepDataCmd::from_str(values[0].trim()) {
+            Ok(val) => val,
+            Err(_) => return Err(make_error("invalid command".to_string())),
+        };
+        let name = values[1].trim();
+        let namez = CString::new(name).unwrap();
+        let cat = unsafe { SRepCatGetByShortname(namez.as_ptr()) };
+        if cat == 0 {
+            return Err(make_error("unknown category".to_string()));
+        }
+        let (mode, arg1) = match values[2].trim() {
+            "isset" => { (DetectUintMode::DetectUintModeGte, 0)}, // any value means it's set
+            "isnotset" => { (DetectUintMode::DetectUintModeEqual, 255) }, // impossible value means not set
```

Comment:
Another solution would be to create `DetectUintModeAny` and `DetectUintModeNone` ?

Or to have the wrapper structure `DetectIPRepData` use booleans `justest` and `isnotset`

Reply:
`DetectUintModeGt` 255 is an invalid setting for u8, even if it is perhaps not enforced anywhere.

Would like 255 to explicit, or even use a named constant to make it easy to understand what it signifies.

Reply:
I guess `DetectUintModeAny` could make some sense, though it could just be conceptualized as `true`, but less sure about `DetectUintModeNone`. I think your suggestion of not overloading the uint support is probably the cleanest.

Reply:
> have the wrapper structure DetectIPRepData use booleans justest and isnotset

Looks cleanest indeed